### PR TITLE
remove enabled components in object caches when exec prune

### PR DESCRIPTION
**Pull request by tuhao1020**
Signed-off-by: HaoTu <haotu@alauda.io>

Please provide a description of this PR:
flushes all object caches when exec prune, fix for https://github.com/istio/istio/issues/37828

**To help us figure out who should review this PR, please put an X in all the areas that this PR affects.**

- [ ] Configuration Infrastructure
- [ ] Docs
- [x] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

### CODE REVIEW

File: operator/pkg/helmreconciler/prune.go [hunk 1]
```diff
@@ -173,6 +173,18 @@ func (h *HelmReconciler) PruneControlPlaneByRevisionWithController(iopSpec *v1al
...
 		}
+
+		crHash, err := h.getCRHash(c)
+		if err != nil {
+			return errStatus, err
+		}
+		scope.Infof("reset cache: %s for CR %s", c, crHash)
+		objCache := cache.GetCache(crHash)
+		if len(objCache.Cache) > 0 {
+			objCache.Mu.Lock()
```
Comment (howardjohn) [inline operator/pkg/helmreconciler/prune.go:184]: nevr unlocked?

Comment (tuhao1020) [reply operator/pkg/helmreconciler/prune.go:184]: done

**Comment by zirain** [comment]
kindly ping @istio/wg-environments-maintainers , this will also fix https://github.com/istio/istio/issues/40903

**Comment by zirain** [comment]
not stale

**Review by zirain — CHANGES_REQUESTED** [review]
I think this's not right after https://github.com/istio/istio/pull/41449 merged.

you should add `tuhao-test` into env `WATCH_NAMESPACE`

env:
          - name: WATCH_NAMESPACE
            value: istio-system,tuhao-test

**Comment by tuhao1020** [comment]
> I think this's not right after #41449 merged.
> 
> you should add `tuhao-test` into env `WATCH_NAMESPACE`
> 
> env: - name: WATCH_NAMESPACE value: istio-system,tuhao-test

Where does the WATCH_NAMESPACE environment variable need to be placed?

**Pull request closed without merging**